### PR TITLE
Issue #2449261 by daniel_j, pcho, joseph.olstad, DamienMcKenna, David…

### LINKED ISSUE #4267: Fix PHP Fatal error: Cannot create references to/from string offsets. (date module)
The below was copied from https://github.com/backdrop/backdrop-issues/issues/4027 because I'd like to get the fatal error fixed, and in order to do so we need an issue for the patch :) 

> Today I ran into this issue that has been resolved in the Drupal date module, so I decided to create this meta to get caught up with Date module for Backdrop. I've also filed a PR for only that issue so that I could get my site working again. I hope to have time to address some of these other changes soon too!

Original Drupal issue: https://www.drupal.org/project/date/issues/2449261

**How to reproduce**

- Create a page view of nodes (details don't matter)
- Add an exposed filter on a node date field (like "created")
- Visit that page, enter incomplete data - that's nice, validation strikes
- Now be a hog and override the $_GET values (browser addressbar) with something unusable, like `dateviewsexposed?date_filter[value]foobar`

**You get**

A bunch of errors and `Cannot create references to/from string offsets `.

A screenshot:
illegal-string-offset

---

PR: https://github.com/backdrop/backdrop/pull/2868

### PR DESCRIPTION
**Pull request by jenlampton**
Part of https://github.com/backdrop/backdrop-issues/issues/4027
Fixes https://github.com/backdrop/backdrop-issues/issues/4267

### CODE REVIEW

File: core/modules/date/views/date_views_filter_handler_simple.inc [hunk 1]
```diff
@@ -283,9 +283,10 @@ class date_views_filter_handler_simple extends views_handler_filter_date {
    */
   function value_form(&$form, &$form_state) {
     // We use different values than the parent form, so we must
-    // construct our own form element.
+    // Construct our own form element.
     $form['value'] = array();
     $form['value']['#tree'] = TRUE;
+    $form['value']['#id'] = 'date_views_exposed_filter-' . bin2hex(drupal_random_bytes(16));
```
Comment (indigoxela) [inline core/modules/date/views/date_views_filter_handler_simple.inc]: Oops, a "drupal_" sneaked in here. ;-) Should be backdrop_random_bytes instead.

Comment (indigoxela) [reply core/modules/date/views/date_views_filter_handler_simple.inc]: Another thought: wouldn't backdrop_random_key() be an easier solution here?

File: core/modules/date/views/date_views_filter_handler_simple.inc [hunk 2]
```diff
@@ -283,9 +283,10 @@ class date_views_filter_handler_simple extends views_handler_filter_date {
    */
   function value_form(&$form, &$form_state) {
     // We use different values than the parent form, so we must
-    // construct our own form element.
+    // Construct our own form element.
```
Comment (klonos) [inline core/modules/date/views/date_views_filter_handler_simple.inc]: Why this??

**Comment by jenlampton** [comment]
closing for new sandbox.

**Comment by jenlampton** [comment]
new sandbox

File: core/modules/date/views/date_views_filter_handler_simple.inc (hunk 2 above)
Comment (quicksketch) [inline core/modules/date/views/date_views_filter_handler_simple.inc]:
```suggestion
    // construct our own form element.
```

Comment (jenlampton) [reply core/modules/date/views/date_views_filter_handler_simple.inc]: Coding standards (Start with uppercase)

Comment (indigoxela) [reply core/modules/date/views/date_views_filter_handler_simple.inc]: @jenlampton that's in the *middle of a sentence*. :smile: " so we must Construct"

**Comment by laryn** [comment]
Sorry I bungled up the commit message, but thanks to @jenlampton, @klonos, @quicksketch, @indigoxela, and @herbdool for the work on this one, building on the D7 patch by daniel_j, pcho, joseph.olstad, David Hernández, collinhaines, vinmassaro, and SkyNet Costa Rica. (Finally) merged into `1.x` and `1.26.x`.

**Pull request merged**